app/assets/javascripts/users.js.erb

```diff
@@ -50,6 +40,37 @@ function initMap() {
   }
 }
 
+function displayTargets() {
+  $.get('/user_targets').done(function(data) {
+    $.each(data.user_targets, function(i, val) {
+      var targetPos = {
+        lat: val['lat'],
+        lng: val['lng']
+      }
+      showTarget(targetPos,val['radius'],val['topic'])
+    })
+  })
+}
+
+function showTarget(position, radius, topic){
+  new google.maps.Circle({
+    strokeWeight: 0,
+    fillColor: targetColor,
+    fillOpacity: 0.70,
+    map: map,
+    center: position,
+    radius: radius
+  })
+  new google.maps.Marker({
+    position: position,
+    map: map,
+    icon: { url: "/assets/"+topic+".png",
```

Comment:
You need to send the path along with the other info. That way you will only handle the path on the server side. Also, this wouldn't work if it's uploaded to a production or staging environment for example.